source/_cookbook/SSL guide.md

```diff
@@ -0,0 +1,570 @@
+
+Did it work? Super!
+
+You now have a remotely accesible HA instance that has a text-based URL and will not drop out if your service provider changes your IP.  But, it is only as secure as the password you set, which can be snooped during your session by a malicious hacker with relative ease.  So we need to set up some encryption with SSL, read on to find out how.
+
+#### {% linkable_title 4 - Obtain an ssl certificate from LetsEncrypt %}
+
+First we need to set up another port forward like we did in step 2.  Set your new rule to:  
+
+```text
+Service name - ha_letsencrypt
+Port Range - 80
+Local IP - YOUR-HA-IP
+Local Port - 80
+Protocol - Both
+```
+
+Remember to save the new rule.
+
+<p class='note'>
+In cases where your ISP blocks port 80 you will need to change the port forward options to forward port 443 from outside to port 443 on your HA device.  Please note that this will limit your options for automatically renewing the certificate, but this is a limitation because of your ISP setup and there is not a lot we can do about it!
+</p>
+
+now SSH in to the device your HA is running on.
+
+
+<p class='note'>
+If you're running the 'standard' setup on a raspberry pi the chances are you just logged in as the 'pi' user.  If not, you may have logged in as the HA user.  There are commands below that require the HA user to be on the sudoers list.  If you are not using the 'standard' pi setup it is presumed you will know how to get your HA user on the sudoers list before continuing.  If you are running the 'standard' pi setup, from your 'pi' user issue the following command (where <hass> is the HA user):
+
+```bash
+$ sudo adduser <hass> sudo
+```
+
+</p>
+
+If you did not already log in as the user that currently runs HA, change to that user (usually hass or homeassistant - you may have used a command similar to this in the past):
+
+```bash
+$ su -s /bin/bash hass 
+```
+
+Make sure you are in the home directory for the HA user:
+
+```bash
+$ cd
+```
+
+We will now make a directory for the certbot software, download it and give it the correct permissions:
+
+```text
+$ mkdir certbot
+$ cd certbot/
+$ wget https://dl.eff.org/certbot-auto
+$ chmod a+x certbot-auto
+```
+
+Now we will run the certbot program to get our ssl certificate.  You will need to include your email address and your duckDNS url in the appropriate places:
+
+```text
+$ ./certbot-auto certonly --standalone --preferred-challenges http-01 --email your@email.address -d examplehome.duckdns.org
+```
+
+Once the program has run it will generate a certificate and other files and place them in a folder `/etc/letsencrypt/` .
+
+Confirm this file has been populated:
+
+```bash
+$ ls /etc/letsencrypt/live/
+```
+
+This should show a folder named exactly after your DuckDNS url.
+
+Our HA user needs access to files within the letsencrypt folder, so issue the following commands to change the permissions.
+
+```bash
+$ sudo chmod 755 /etc/letsencrypt/live/

```

Comment:
In all honesty I have no idea!  I am not so familiar with the unix permissions system tbh, and I think I've only ever used 755 and 777.  Looking it up I guess 744 would be enough, but I haven't tested it.